app/models/user.rb

```diff
@@ -14,4 +14,32 @@ def sanitize_phone_number
     phone_number.gsub(/-/, '').prepend("+1")
   end
 
+  def personal_weekly_progress(week_id)
+     goals.where(week_id: week_id).map {|goal| goal.progress_count }.reduce(:+)
+  end
+  def personal_weekly_total_achieved(week_id)
```

Comment:
also, you're passing "week_id" into almost every method here. is there a way to break these methods down so we don't have to pass week_id into every method?